src/components/VideoCard/VideoCard.component.jsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { Grid, Paper, Divider, Button } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import { useFavorites } from '../../FavoritesContext'
+import { useHistory } from 'react-router-dom'
+import { Link } from 'react-router-dom';
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    padding: '1em',
+    maxWidth: '250px',
+    minWidth: '250px',
+    height: 230,
+    backgroundColor: '#cfd8dc',
+  },
+
+  paperFavorite: {
+    padding: '1em',
+    maxWidth: '250px',
+    minWidth: '250px',
+    height: 230,
+    backgroundColor: '#F0B27A',
+  },
+
+  paperTitle: {
+    textOverflow: 'ellipsis',
+    overflow: 'hidden',
+  },
+
+  thumbnail: {
+    width: '250px',
+  },
+
+  button: {
+    margin: theme.spacing(1),
+    
+  },
+}));
+
+function VideoCard(props) {
+  const classes = useStyles();
+  const history =useHistory()
+  const { favorites, addFavorite, removeFavorite } = useFavorites()
+  let isFavorite = props.isFavorite
+
+  const goToWatch = () => {
+      history.push(`/watch/${props.id}`)
+  }
+
+  let favorite_Button;
+
+  if (isFavorite) {
+    favorite_Button = <Button onClick = {()=> addFavorite(props.video)} className={classes.button} size="small" variant="contained" color="primary">Remove</Button>;
+  } else {
+    favorite_Button = <Button onClick = {()=> addFavorite(props.video)} className={classes.button} size="small" variant="contained" color="primary">Favorite</Button>;
+  }
```

Comment:
A ternary it's simpler than declaring the whole component again:

```suggestion
 favorite_Button = <Button onClick = {()=> addFavorite(props.video)} className={classes.button} size="small" variant="contained" color="primary">{isFavorite ? 'Favorite' : 'Remove'}</Button>;
```